src/main/java/net/minestom/server/extensions/ExtensionManager.java

```diff
@@ -216,6 +216,18 @@ private Extension attemptSingleLoad(@NotNull DiscoveredExtension discoveredExten
             LOGGER.error("Main class '{}' in '{}' has no logger field.", mainClass, extensionName, e);
         }
 
+        // Set dataDirectory
+        try {
+            Field dataDirectoryField = Extension.class.getDeclaredField("dataDirectory");
```

Comment:
This could live in the origin - would enable others to easily get the data folder for an extension and also reduce the amount of reflection we do.